src/mumble/GlobalShortcut_win.cpp

```diff
@@ -130,576 +142,395 @@ GlobalShortcutWin::~GlobalShortcutWin() {
 }
 
 void GlobalShortcutWin::run() {
-	if (HAS_FAILED(DirectInput8Create(GetModuleHandle(nullptr), DIRECTINPUT_VERSION, IID_IDirectInput8,
-									  reinterpret_cast< void ** >(&pDI), nullptr))) {
-		qFatal("GlobalShortcutWin: Failed to create d8input");
-		return;
-	}
-
-	// Print the user's LowLevelHooksTimeout registry key for debugging purposes.
-	// On Windows 7 and greater, Windows will silently remove badly behaving hooks
-	// without telling the application. Users can tweak the timeout themselves
-	// with this registry key.
-	HKEY key    = nullptr;
-	DWORD type  = 0;
-	DWORD value = 0;
-	DWORD len   = sizeof(DWORD);
-	if (RegOpenKeyExA(HKEY_CURRENT_USER, "Control Panel\\Desktop", 0, KEY_READ, &key) == ERROR_SUCCESS) {
-		LONG err =
-			RegQueryValueExA(key, "LowLevelHooksTimeout", nullptr, &type, reinterpret_cast< LPBYTE >(&value), &len);
-		if (err == ERROR_SUCCESS && type == REG_DWORD) {
-			qWarning("GlobalShortcutWin: Found LowLevelHooksTimeout with value = 0x%lx",
-					 static_cast< unsigned long >(value));
-		} else if (err == ERROR_FILE_NOT_FOUND) {
-			qWarning("GlobalShortcutWin: No LowLevelHooksTimeout registry key found.");
-		} else {
-			qWarning(
-				"GlobalShortcutWin: Error looking up LowLevelHooksTimeout. (Error: 0x%lx, Type: 0x%lx, Value: 0x%lx)",
-				static_cast< unsigned long >(err), static_cast< unsigned long >(type),
-				static_cast< unsigned long >(value));
+#ifdef USE_XBOXINPUT
+	if (Global::get().s.bEnableXboxInput) {
+		m_xinput.reset(new XboxInput());
```

Comment:
Technically it might be a bit cleaner to do something like
```suggestion
		m_xinput = std::make_unique<XboxInput>();
```
(For the sake of exception safety)

The same thing applies to other instances in which you use `reset` to assign a new object to the pointer.